src/frontend/components/UI/WebviewControls/index.tsx

```diff
@@ -66,6 +68,10 @@ export default function WebviewControls({
           return webview?.reload()
         }
         if (event === 'back') {
+          // if the user is on the first page, offset -1 will be hyperplay store
+          // and -2 will be the undefined page
+          const isFirstPage = webview?.canGoToOffset(-2)
+          if (!isFirstPage) return navigate('/')
```

Comment:
this will fail if we:
1. go to hp store
2. go to epic store
3. go to gog
4. click back which will take us to epic store but gog will still be highlighted since we never navigated there

we need to listen to the webview element for when it navigates and use that url 

adding the following to src/frontend/screens/WebView/index.tsx will work
```
webview.addEventListener('did-navigate', (ev) =>
        console.log('did navigate ', ev.url)
      )
```

Reply:
maybe instead of using the pathname and search params in `TopNavBar` we could use a mobx store that is updated with the url that the webview is navigated to?

Reply:
Oh that’s a great call, we could use persisted value to render the styles only.